crates/fuel-core/src/executor.rs

```diff
@@ -691,101 +691,171 @@ where
-                Err(ExecutorError::NotSupportedTransaction(*tx_id))
-            }
+            CheckedTransaction::Mint(mint) => self.execute_mint(
+                mint,
+                header,
+                execution_data,
+                tx_db_transaction,
+                execution_kind,
+                options,
+            ),
         }
     }
 
-    fn apply_coinbase(
+    fn execute_mint(
         &self,
-        coinbase_tx: Mint,
-        block: &PartialFuelBlock,
+        checked_mint: Checked<Mint>,
+        header: &PartialBlockHeader,
         execution_data: &mut ExecutionData,
         block_db_transaction: &mut DatabaseTransaction,
-    ) -> ExecutorResult<()> {
-        let block_height = *block.header.height();
-        let coinbase_id = coinbase_tx.id(&self.config.consensus_parameters.chain_id);
-        self.persist_output_utxos(
-            block_height,
-            0,
-            &coinbase_id,
-            block_db_transaction,
-            &[],
-            coinbase_tx.outputs(),
-        )?;
-        execution_data.tx_status.insert(
-            0,
-            TransactionExecutionStatus {
-                id: coinbase_id,
-                result: TransactionExecutionResult::Success { result: None },
-            },
-        );
-        if block_db_transaction
-            .deref_mut()
-            .storage::<Transactions>()
-            .insert(&coinbase_id, &coinbase_tx.into())?
-            .is_some()
-        {
-            return Err(ExecutorError::TransactionIdCollision(coinbase_id))
-        }
-        Ok(())
-    }
+        execution_kind: ExecutionKind,
+        options: ExecutionOptions,
+    ) -> ExecutorResult<Transaction> {
+        execution_data.found_mint = true;
 
-    fn check_coinbase(
-        &self,
-        block_height: BlockHeight,
-        mint: Mint,
-        expected_amount: Option<Word>,
-    ) -> ExecutorResult<Mint> {
-        let checked_mint =
-            mint.into_checked(block_height, &self.config.consensus_parameters)?;
-
-        if checked_mint.transaction().tx_pointer().tx_index() != 0 {
-            return Err(ExecutorError::CoinbaseIsNotFirstTransaction)
+        if checked_mint.transaction().tx_pointer().tx_index() != execution_data.tx_count {
+            return Err(ExecutorError::MintHasUnexpectedIndex)
         }
 
-        if checked_mint.transaction().outputs().len() > 1 {
-            return Err(ExecutorError::CoinbaseSeveralOutputs)
-        }
+        let coinbase_id = checked_mint.id();
+        let (mut mint, _) = checked_mint.into();
 
-        if let Some(Output::Coin {
-            asset_id, amount, ..
-        }) = checked_mint.transaction().outputs().first()
-        {
-            if asset_id != self.config.consensus_parameters.base_asset_id() {
-                return Err(ExecutorError::CoinbaseOutputIsInvalid)
+        if mint.input_contract().contract_id != ContractId::zeroed() {
+            if *mint.mint_amount() != execution_data.coinbase {
```

Comment:
I think it would improve readability a lot to break down these functions into sub-functions. These functions are long.

Reply:
This is an improvement. I don't know if I'm a fan of the function being in shared scope, it is about as cluttered as before. Also, having the "else" case be encapsulated as a function wouldn't hurt either, but I don't want to create too much extra work for you :). Clean Code suggests avoiding functions that are more than 20 lines long, that is an arbitrary number, but I tend to agree in spirit--breaking down functions makes a world of difference on cognitive load for a reader.

Reply:
I don't want to add one more function that accepts 8 arguments=)

Reply:
Okay. I'll leave it as a nit, however I think we shouldn't let something like that keep us from making our code more friendly and sympathetic to the next engineer.